### PR TITLE
feat(b-overlay): add support for overlay `click` event (closes #5243)

### LINKED ISSUE #5243: Support click handler on <b-overlay /> shade.
### Is your feature request related to a problem? Please describe...

I am looking for a way to be able to trigger a callback when a  `<b-overlay />` is clicked. This is useful when using the overlay to cover the entire screen to have the user focus on a specific piece of content.

### Describe the solution you'd like

I would like to be able to add a click handler to the overlay using `@click="..."`.

```vue
<b-overlay
    @click="closeOverlay"
>
    ...
</b-overlay>
```

### Describe alternatives you've considered

I've considered adding a close button to the content shown in the overlay, however it is a common UX pattern to allow a user to close an overlay by clicking on the shaded portion.

### Additional context

I'm new to Vue so adding such a click handler may already be supported without changes to the library. Please let me know if this is possible, thanks :).

### PR DESCRIPTION
**Pull request by jackmu95**
### Describe the PR

This PR adds support for the `click` event on the `<b-overlay>` component's overlay.

### PR checklist

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix (fixes a boo-boo in the code) - `fix(...)`, requires a patch version update
- [x] Feature (adds a new feature to BootstrapVue) - `feat(...)`, requires a minor version update
- [ ] Enhancement (augments an existing feature) - `feat(...)`, requires a minor version update
- [ ] ARIA accessibility (fixes or improves ARIA accessibility) - `fix(...)`, requires a patch or minor version update
- [ ] Documentation update (improves documentation or typo fixes) - `chore(docs)`, requires a patch version update
- [ ] Other (please describe)

**Does this PR introduce a breaking change?** (check one)

- [x] No
- [ ] Yes (please describe since breaking changes require a minor version update)

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch, **not** the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (i.e. `[...] (fixes #xxx[,#xxx])`, where "xxx" is the issue number)
- [x] It should address only one issue or feature. If adding multiple features or fixing a bug and adding a new feature, break them into separate PRs if at all possible.
- [x] The title should follow the [**Conventional Commits**](https://www.conventionalcommits.org/) naming convention (i.e. `fix(alert): not alerting during SSR render`, `docs(badge): update pill examples`, `chore(docs): fix typo in README`, etc). **This is very important, as the `CHANGELOG` is generated from these messages, and determines the next version type (patch or minor).**

**If new features/enhancement/fixes are added or changed:**

- [ ] Includes documentation updates
- [ ] Includes component package.json meta section updates (prop, slot and event changes/updates)
- [ ] Includes any needed TypeScript declaration file updates
- [ ] New/updated tests are included and passing (required for new features and enhancements)
- [ ] Existing test suites are passing
- [ ] CodeCov for patch has met target (all changes/updates have been tested)
- [ ] The changes have not impacted the functionality of other components or directives
- [ ] ARIA Accessibility has been taken into consideration (Does it affect screen reader users or keyboard only users? Clickable items should be in the tab index, etc.)

**If adding a new feature, or changing the functionality of an existing feature, the PR's
description above includes:**

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

### CODE REVIEW

File: src/components/overlay/overlay.js [hunk 1]
```diff
@@ -146,7 +146,8 @@ export const BOverlay = /*#__PURE__*/ Vue.extend({
             'position-absolute': !this.noWrap || (this.noWrap && !this.fixed),
             'position-fixed': this.noWrap && this.fixed
           },
-          style: { ...positionCover, zIndex: this.zIndex || 10 }
+          style: { ...positionCover, zIndex: this.zIndex || 10 },
+          on: { click: evt => this.$emit('click', evt) }
```
Comment (tmorehouse) [inline src/components/overlay/overlay.js:150]: Maybe this should be `overlay-click`?

Comment (jackmu95) [reply src/components/overlay/overlay.js:150]: thought about that too but I think it is fine like that.

Comment (tmorehouse) [reply src/components/overlay/overlay.js:150]: Vue 3 is going to make it difficult to use custom events names that are the same as native event names (they got rid of the `.native` modifier)

Comment (tmorehouse) [reply src/components/overlay/overlay.js:150]: as all future events will be added by `onClick="handler"` (i.e. as `$attrs`) where there is no option to bind a native listener (if users want to) to the root element :crying_cat_face:

**Pull request merged**